docs/topics/scope-functions.md

```diff
@@ -458,7 +458,7 @@ fun main() {
         val hexDigits = "A-Fa-f"
         val sign = "+-"
         
-        Regex("[$sign]?[$digits$hexDigits]+")
+        Regex("[$sign][$digits$hexDigits]+")
```

Comment:
With the ? sign "be" from "not-a-number" is included in the result, see https://kotlinlang.org/docs/scope-functions.html#run

Without ?, the result is what can be expected: https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS41LjIxIiwiY29kZSI6ImZ1biBtYWluKCkge1xuICAgICAgICB2YWwgaGV4TnVtYmVyUmVnZXggPSBydW4ge1xuICAgICAgICAgICAgdmFsIGRpZ2l0cyA9IFwiMC05XCJcbiAgICAgICAgICAgIHZhbCBoZXhEaWdpdHMgPSBcIkEtRmEtZlwiXG4gICAgICAgICAgICB2YWwgc2lnbiA9IFwiKy1cIlxuXG4gICAgICAgICAgICBSZWdleChcIlskc2lnbl1bJGRpZ2l0cyRoZXhEaWdpdHNdK1wiKVxuICAgICAgICB9XG5cbiAgICAgICAgZm9yIChtYXRjaCBpbiBoZXhOdW1iZXJSZWdleC5maW5kQWxsKFwiKzEyMzQgLUZGRkYgbm90LWEtbnVtYmVyXCIpKSB7XG4gICAgICAgICAgICBwcmludGxuKG1hdGNoLnZhbHVlKVxuICAgICAgICB9XG59IiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiJ9

Reply:
* `be` is a valid hex number, so this works as expected now :) 
* If you remove `?`, you won't find numbers without a sign, which is incorrect. Try adding `10` in the string.

Although I totally agree that it's a tricky example, maybe we should replace the string we search with something simpler.

Reply:
> be is a valid hex number, so this works as expected now :)
Yes, but it's  a bit confusing: you expect to see only two numbers as the result but see three including "be")

> If you remove ?, you won't find numbers without a sign, which is incorrect. 
Of course, I understand it)

We can replace "not-a-number" with "!!!%*&". WDYT?

Reply:
Of course, that's what I meant. Or just take some letters that don't form a hex number. And add a simple number without a sign to show that the regex matches them as well.

Reply:
> And add a simple number without a sign to show that the regex matches them as well.

Added.